samples/algorithms/MajoranaQubits/src/GateSet.qs

```diff
@@ -0,0 +1,62 @@
+/// A set of gates built upon the custom measurements
+/// provided by the hardware provider.
+///
+/// Source:
+///  [1] Surface code compilation via edge-disjoint paths
+///      https://arxiv.org/pdf/2110.11493
+
+/// Apply a CNOT gate to the given qubits.
+/// Source: [1] Figure 3.
+operation CNOT(control : Qubit, target : Qubit) : Unit {
+    use ancilla = Qubit();
+    let a = Mzz(control, target);
+    let b = Mxx(target, ancilla);
+    let c = Mx(target);
+    if b == One {
+        Z(control);
+    }
+    if [a, c] == [One, One] {
+        X(ancilla);
```

Comment:
Not sure if we need some explanatory comments here (or it we expect folks have read the paper), but what's the point of doing an X on 'ancilla' here? It's about to go out of scope and nothing further is done with it in this operation.

Reply:
My bad. The ancilla qubit was the middle one in the diagram. I read it wrong.